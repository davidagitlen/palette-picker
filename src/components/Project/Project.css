label {
  font-size: 18px;
}

.project-wrapper {
  flex-direction: column;
  margin: 35px auto 0 auto;
  width: 40%;
}

.project-wrapper > h3 {
  font-family: 'Source Code Pro', monospace;
  background-color: #424A60;
  color: snow;
  font-size: 28px;
  height: 50px;
  line-height: 45px;
  margin: 0;
  padding: 0 0 0 10px;
}

.project-name-input {
  box-shadow: 3px 3px 7px #A9A9A9;
  font-size: 21px;
  height: 50px;
  margin: 0px 0px 25px 0px;
  padding: 0px 0px 0px 5px;
  text-align: left;
  width: 98%;
}

.project-name-input::placeholder {
  color: black;
}

.Project {
  align-content: flex-start;
  background-color: rgba(240,240,240, 0.7);
  box-shadow: 1px 1px 4px #D3D3D3;
  display: flex;
  flex-direction: row;
  height: max-content;
  margin: 25px auto;
  min-height: 150px;
  width: 90%;
}

.Project > input {
  width: 50px;
  height: 50px;
}

.input-wrapper {
  display: flex;
  flex-direction: column;
  margin: 35px 0px 0px 0px;
}

.one-to-many {
  height: 80px;
  padding: 60px 0 0 0;
  width: 100px;
}

#trash,
#update {
  display: inline-block;
  height: 30px;
  width: 100px;
}

label:hover,
#trash:hover,
#update:hover {
  border: none;
  cursor: pointer;
  text-decoration: underline #424A60;
}