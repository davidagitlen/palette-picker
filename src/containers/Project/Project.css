label {
  font-size: 18px;
}

.project-wrapper {
  flex-direction: column;
  margin: 25px auto 0 auto;
  width: 40%;
}

.project-wrapper > h3 {
  background-color: #424A60;
  color: snow;
  font-size: 28px;
  height: 50px;
  line-height: 45px;
  padding: 2px 0 0 5px;
  vertical-align: center;
}

.project-name-input {
  text-align: left;
  height: 50px;
  margin: 0px 0px 25px 0px;
  padding: 0px 0px 0px 5px;
}

.project-name-input::placeholder {
  color: black;
}

.Project {
  align-content: flex-start;
  background-color: #E8E8E8;
  display: flex;
  flex-direction: row;
  height: max-content;
  margin: 25px auto;
  min-height: 150px;
  width: 90%;
}

.Project > input {
  width: 50px;
  height: 50px;
}

.input-wrapper {
  display: flex;
  flex-direction: column;
  margin: 35px 0px 0px 0px;
}

.cross {
  height: 50px;
  padding: 60px 0 0 0;
  width: 100px;
}

#trash,
#update {
  display: inline-block;
  height: 30px;
  width: 100px;
}