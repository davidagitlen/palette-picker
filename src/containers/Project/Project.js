import React, { Component } from 'react';
import './Project.css';
import Palettes from '../Palettes/Palettes';
import update from '../../images/update.svg';
import trash from '../../images/trash.svg';
import oneToMany from "../../images/one_to_many.png";

class Project extends Component{
  constructor(props){
    super(props);
    this.state = {
      project: ''
    }
  }

  handleChange = e => {
    this.setState({project: e.target.value}, () => {
      console.log(this.state)
    })
  }
  render() {
    const { 
      project, 
      palettes, 
      editProject, 
      editPalette, 
      trashProject, 
      trashPalette,
      selectPalette } = this.props;
    const filteredPalettes = palettes.filter(palette => palette.project_id === project.id)
    return (
      <article className="Project">
        <div className="input-wrapper">
          <input
            id="trash"
            type="image"
            alt="trash"
            src={trash}
            onClick={() => trashProject(project.id)}
          />
          <label htmlFor="trash">Trash</label>
        </div>
        <div className="input-wrapper">
          <input
            id="update"
            type="image"
            alt="update"
            src={update}
            onClick={() => editProject(this.state.project, project.id)}
          />
          <label htmlFor="update">Update</label>
        </div>
        <div className="project-wrapper">
          <h3>Project:</h3>
          <input
            className="project-name-input"
            type="text"
            value={this.state.project}
            placeholder={project.project}
            onChange={this.handleChange}
          />
        </div>
        <img src={oneToMany} alt="line" className="one-to-many" />
        <Palettes
          palettes={filteredPalettes}
          editPalette={editPalette}
          trashPalette={trashPalette}
          selectPalette={selectPalette}
        />
      </article>
    );
  }

}

export default Project;