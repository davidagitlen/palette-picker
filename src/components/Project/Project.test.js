import React from 'react';
import { shallow } from 'enzyme';
import Project from './Project';

it('initial test to pass CI suite', () => {
  expect(true).toEqual(true);
});

describe('Project', () => {

  let wrapper, project, palettes, mockEditProject, mockEditPalette, mockTrashProject, mockTrashPalette, mockSelectPalette;

  beforeEach(() => {
    project = { id: 1, project: 'Project_One' };
    palettes = [
      {
        hex_1: "#3192EC",
        hex_2: "#77B8AB",
        hex_3: "#910E31",
        hex_4: "#D8C86A",
        hex_5: "#6E16C2",
        id: 34,
        palette: "one",
        project_id: 1,
        project_name: "Blah blah"
      },
      {
        hex_1: "#3192EC",
        hex_2: "#77B8AB",
        hex_3: "#910E31",
        hex_4: "#D8C86A",
        hex_5: "#6E16C2",
        id: 34,
        palette: "two",
        project_id: 2,
        project_name: "Blib blob"
      }
    ];
    mockEditProject = jest.fn();
    mockEditPalette = jest.fn();
    mockTrashProject = jest.fn();
    mockTrashPalette = jest.fn();
    mockSelectPalette = jest.fn();
    wrapper = shallow(
      <Project 
        project={project}
        palettes={palettes}
        editProject={mockEditProject}
        editPalette={mockEditPalette}
        trashProject={mockTrashProject}
        trashPalette={mockTrashPalette}
        selectPalette={mockSelectPalette}
      />
    );
  });

  it('should match the snapshot with data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke trashProject with the project id passed through when a click is detected on the trash svg', () => {

    wrapper.find('#trash').simulate('click');

    expect(mockTrashProject).toHaveBeenCalledWith(project.id);
  });

  it('should invoke editProject with the current project and project id passed through when a click is detected on the update svg', () => {

    wrapper.instance().setState({project});

    wrapper.find('#update').simulate('click');

    expect(mockEditProject).toHaveBeenCalledWith(project, project.id);
  });

  describe('handleChange', () => {
    it('should update the project property in state with the event target value', () => {
      const mockEvent = {
        target: {
          value: 'Test_Project'
        }
      };
      const firstExpected = {
        project: ''
      }
      const expected = {
        project: 'Test_Project'
      }

      expect(wrapper.state()).toEqual(firstExpected);

      wrapper.find('.project-name-input').simulate('change', mockEvent);

      expect(wrapper.state()).toEqual(expected);
    });
  });


});