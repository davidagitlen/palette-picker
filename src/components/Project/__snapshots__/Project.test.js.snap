// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should match the snapshot with data passed in correctly 1`] = `
<article
  className="Project"
>
  <div
    className="input-wrapper"
  >
    <input
      alt="trash"
      id="trash"
      onClick={[Function]}
      src="trash.svg"
      type="image"
    />
    <label
      htmlFor="trash"
    >
      Trash
    </label>
  </div>
  <div
    className="input-wrapper"
  >
    <input
      alt="update"
      id="update"
      onClick={[Function]}
      src="update.svg"
      type="image"
    />
    <label
      htmlFor="update"
    >
      Update
    </label>
  </div>
  <div
    className="project-wrapper"
  >
    <h3>
      Project:
    </h3>
    <input
      className="project-name-input"
      onChange={[Function]}
      placeholder="Project_One"
      type="text"
      value=""
    />
  </div>
  <img
    alt="line"
    className="one-to-many"
    src="one_to_many.png"
  />
  <Palettes
    editPalette={[MockFunction]}
    palettes={
      Array [
        Object {
          "hex_1": "#3192EC",
          "hex_2": "#77B8AB",
          "hex_3": "#910E31",
          "hex_4": "#D8C86A",
          "hex_5": "#6E16C2",
          "id": 34,
          "palette": "one",
          "project_id": 1,
          "project_name": "Blah blah",
        },
      ]
    }
    selectPalette={[MockFunction]}
    trashPalette={[MockFunction]}
  />
</article>
`;
